### PR TITLE
FIX Issue 4082: ComboBox wrong border and background colors for dropdown button with forced-colors: active

### PR DESCRIPTION
**Pull request by jcpengy10**
Closes (https://github.com/adobe/react-spectrum/issues/4082)

## ✅ Pull Request Checklist:

- [ x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ x] Filled out test instructions.
- [ x] Updated documentation (if it already exists for this component).
- [ x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Desktop: Textfield and button are both highlighted 
Screen Shot 2023-04-28 at 4 31 07 PM

Mobile: Textfield and button are both green on disabled
Screen Shot 2023-04-28 at 4 31 23 PM

Express:
Screen Shot 2023-04-28 at 4 31 54 PM

Express Mobile:
Screen Shot 2023-04-28 at 4 32 09 PM

## 🧢 Your Project:

Adobe

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 1]
```diff
@@ -202,12 +208,16 @@ governing permissions and limitations under the License.
     /* Focus ring: When one of the inputs or the button has keyboard focus, render the focus ring border around the entire input group by styling an adjacent descendant element. */
     &:focus-ring {
       .spectrum-FieldButton {
-        border-color: var(--spectrum-dropdown-border-color-key-focus);
-        box-shadow: 0 0 0 1px var(--spectrum-dropdown-border-color-key-focus);
+        &.before {
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:211]: this goes for all of them
```suggestion
        &:before {
```

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 2]
```diff
@@ -202,12 +208,16 @@ governing permissions and limitations under the License.
...
+           border-color: var(--spectrum-dropdown-border-color-key-focus);
+           box-shadow: 0 0 0 1px var(--spectrum-dropdown-border-color-key-focus);
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:213]: not sure box-shadow should be included in the `:before`, lets leave it on the FieldButton. the `:before` should only encompass border-color

**Comment by snowystinger** [comment]
GET_BUILD

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 3]
```diff
@@ -258,8 +270,10 @@ governing permissions and limitations under the License.
     }
 
     &.is-disabled:not(.spectrum-InputGroup--invalid):not(.spectrum-InputGroup--quiet) .spectrum-FieldButton {
-      border-color: GrayText;
       border-width: 1px;
+      &:before {
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:274]: I think we need to delete the border-width on the line above this as well
See https://reactspectrum.blob.core.windows.net/reactspectrum/c66b7154aec76fef87da719f58afbc190fea70a3/storybook/index.html?path=/story/combobox--default&args=isDisabled:true&providerSwitcher-express=false:
Screenshot 2023-05-03 at 11 46 50 AM


VS if we turn off that property
Screenshot 2023-05-03 at 11 47 10 AM

**Comment by snowystinger** [comment]
GET_BUILD

**Review by snowystinger — APPROVED** [review]

**Review by ktabors — APPROVED** [review]

**Pull request merged**